src/common/main.lua

```diff
@@ -54,6 +58,44 @@ main.init = function()
 			Data = json.encode(utils.affiliationsForAddress(address, ANTS)),
 		})
 	end)
+
+	utils.createActionHandler(ActionMap.AddVersion, function(msg)
+		assert(msg.From == Owner, "Only ANT Registry owner can add versions")
+		local version = tonumber(msg.Version)
+		local moduleId = msg["Module-Id"]
+		local luaSourceId = msg["Lua-Source-Id"]
+		local notes = msg.Notes or ""
+
+		assert(
+			version ~= nil and math.type(version) == "integer" and version >= 0,
+			"Version must be a positive integer, recieved " .. tostring(version)
+		)
+		utils.validateArweaveId(moduleId)
+		assert(
+			type(luaSourceId) == "string" and utils.validateArweaveId(luaSourceId) or luaSourceId == nil,
+			"Lua-Source-Id should be a valid arweave ID"
+		)
+		assert(type(notes) == "string", "Notes must be a string")
+
+		ANTVersions[tostring(version)] =
```

Comment:
reason for the tostring here for storage, is lua complains about having a "sparse array", an table in which there are nil spaces in between indicies. We can work around this by indexing with strings, or using an arraylike structure and using table.insert

Reply:
ANTVersions was to be just an array, can you insert at specific indices?